variables-workers.tf

```diff
@@ -0,0 +1,232 @@
+  type        = string
+}
+
+variable "cluster_dns" {
+  default     = "10.96.5.5"
+  description = "Cluster DNS resolver IP address. The provided value used with default 'servicesCidr' 10.96.0.0/16 should only require modification in the case of conflict."
+  type        = string
+}
+
+#
+# Worker pools
+#
+
+variable "worker_pools" {
+  default     = {}
+  description = "Tuple of OKE worker pools where each key maps to the OCID of an OCI resource, and value contains its definition."
+  type        = any
+}
+
+variable "worker_pool_enabled" {
+  default     = true
+  description = "Whether to apply resources for a group when unspecified."
+  type        = bool
+}
+
+variable "worker_pool_mode" {
+  default     = "node-pool"
+  description = "Default management mode for worker pools when unspecified. Only 'node-pool' is currently supported."
+  type        = string
+  validation {
+    condition     = contains(["node-pool", "instance-pool", "cluster-network"], var.worker_pool_mode)
+    error_message = "Accepted values are node-pool, instance-pool, or cluster-network"
+  }
+}
+
+variable "worker_pool_size" {
+  default     = 0
+  description = "Default size for worker pools when unspecified."
+  type        = number
+}
+
+variable "output_worker_detail" {
+  default     = false
+  description = "Whether to include detailed output for worker pool configuration."
+  type        = bool
+}
+
+#
+# Workers: network
+#
+
+variable "worker_type" {
+  default     = "private"
+  description = "Whether to provision public or private workers."
+  type        = string
+  validation {
+    condition     = contains(["public", "private"], var.worker_type)
+    error_message = "Accepted values are public or private."
+  }
+}
+
+variable "worker_nsg_ids" {
+  default     = []
+  description = "An additional list of network security group (NSG) IDs for node security."
+  type        = list(string)
+}
+
+variable "pod_nsg_ids" {
+  default     = []
+  description = "An additional list of network security group (NSG) IDs for pod security."
+  type        = list(string)
+}
+
+variable "kubeproxy_mode" {
+  default     = "iptables"
+  description = "The mode in which to run kube-proxy."
+  type        = string
+
+  validation {
+    condition     = contains(["iptables", "ipvs"], var.kubeproxy_mode)
+    error_message = "Accepted values are iptables or ipvs."
+  }
+}
+
+#
+# Workers: instance
+#
+
+variable "worker_block_volume_type" {
+  default     = "paravirtualized"
+  description = "The default block volume attachment type for Instance Configurations."
+  type        = string
+  validation {
+    condition     = contains(["iscsi", "paravirtualized"], var.worker_block_volume_type)
+    error_message = "Accepted values are 'iscsi' or 'paravirtualized'."
+  }
+}
+
+variable "worker_node_labels" {
```

Comment:
Looked into this - I wouldn't disagree if we were talking about a few variables, but the reality is that anything can be defined at the pool level, so they're _all_ "defaults". The variable descriptions call this out; I'm afraid prefixing the names will make them extremely long without adding much value, since the whole paradigm is to allow override, and behavior is described with each (e.g. merge vs. when unspecified).